puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/nightly_packaging.groovy

```diff
@@ -0,0 +1,42 @@
+def setup_nightly_build_environment(args) {
```

Comment:
This is my limited knowledge of groovy: why do you use args instead of named arguments here?

Reply:
It allows you to pass in key-value pairs when calling.

```groovy
setup_nightly_build_environment(
    git_url: 'https://github.com/theforeman/smart-proxy.git',
    git_ref: 'develop',
    project_name: 'foreman-proxy'
)

I mostly stick with it to keep in line with most other helper methods I've seen in `foreman-infra`.

Reply:
This makes the calling of the method clearer to read but the method signature itself less clearer. We've err'd on the side of making the calling of the method clearer to read to make reading pipelines more obvious.